substrate/frame/revive/src/evm/tracing/prestate_tracing.rs

```diff
@@ -254,31 +275,34 @@ where
 		&mut self,
 		from: H160,
 		to: H160,
-		is_delegate_call: bool,
+		delegate_call: Option<H160>,
 		_is_read_only: bool,
 		_value: U256,
 		_input: &[u8],
 		_gas: Weight,
 	) {
-		if is_delegate_call {
-			self.calls.push(self.current_addr());
+		if let Some(delegate_call) = delegate_call {
+			self.calls.push(Call::DelegateCall(self.current_addr(), delegate_call));
+			self.read_account(delegate_call);
```

Comment:
this is probably the only change we need and we only need to do it in prestate mode